### PR TITLE
Anchor link DEMO

### PR DESCRIPTION
**Pull request by petesfrench**
**As the git method is not available in our demo environment, you will need to run this locally to test it**
*This is just for demoing purposes, I would like to upstream this to vanilla*

## Done

- Add style for discourse module update that allows anchor tags on headings

## QA

- Go to https://microk8s-io-642.demos.haus/docs/howto-addons and hover over one of the h2 headings
- See that it is clickable and has a fun shape next to it
- Click it and see the anchor tag in the url update :exploding_head: 

## Issue / Card

Fixes #

## Screenshots

[if relevant, include a screenshot]

### CODE REVIEW

**Comment by petesfrench** [comment]
@bartaz This is the anchor tags work I mentioned to you via email. I have include the styling locally to this project, but I think it would be better to upstream it to vanilla for better control. We would of course have to make sure all projects that leverage this are on the most recent vanilla before we update the module. What do you think? (the visuals will of course need work)

**Comment by bartaz** [comment]
Focus gets wonky:

image

**Comment by bartaz** [comment]
I'm also wondering on backend side of this.
How will it behave with different kinds of headings from discourse?

I guess ideal situation is when there is only a heading text - but what if there is an anchor placed there manually by editor? (like this: https://juju.is/docs/juju/tutorial#heading--part-1-get-acquainted-with-juju)

(I'm not sure if I haven't seen a different style of this in some other place, don't remember now).
Maybe worth reaching to docs team to ask how are they doing this right now, to take that into account.

**Comment by carkod** [comment]
You can git to the dockerfile and it will work.

**Review by carkod — APPROVED** [review]
LGTM but is there a chance a documentation author doesn't want an anchor? This will force everyone to adhere to having anchors in every h2?

**Comment by juanruitina** [comment]
I think this is a fantastic idea. It works well on my side.

I wouldn't use the paragraph character (¶, or pilcrow as I just learnt) as the icon on hover. It feels unnecessarily obscure. I would suggest to use instead a link icon (🔗) as [Google](https://cloud.google.com/docs/overview) or [GitHub](https://docs.github.com/en/get-started/start-your-journey/about-github-and-git) do. However, we don't seem to have one in Vanilla, am I correct? Could we consider adding it? @bartaz @lyubomir-popov (see below what it looks like now)

Screenshot 2024-03-21 at 17 52 47

In the meantime, we could use `p-icon--anchor` (⚓︎). A bit quirky, but more accurate (and rather cute).

One accessibility question: I see GitHub using `tabindex="-1"`. I wonder if we should do that on the links to exclude them from the tabbing order: if we don't, tabbing through can become unnecessarily cumbersome. I understand it's not really useful for screen reader users and they would navigate through headings intentionally if they wanted so, they won't expect that behaviour with tabbing. If we have a table of contents as is the case, it's even worse because we are making go through headings twice. (Interestingly, GitHub added this to headings, not to the anchor links, so it doesn't have that effect for them)

**Comment by juanruitina** [comment]
@bartaz Thank you for the links, many different approaches, yes!

As for the icon, now that I see it on MDN, I wouldn't oppose just using just "#" instead too, perhaps in $color-mid-dark so it doesn't meld with the actual heading. @lyubomir-popov thoughts?

File: static/sass/styles.scss [hunk 1]
```diff
@@ -153,3 +154,27 @@
...
+    @extend .p-icon--get-link;
+    content: "¶";
+    display: inline-block;
+    padding: 1rem;
+    top: 0.6rem;
+    visibility: hidden;
+    margin-left: 0.75rem;
+  }
+
+  &:hover {
```
Comment (bartaz) [inline static/sass/styles.scss:175]: May be good to make it visible on focus as well, for consistency?

File: static/sass/styles.scss [hunk 2]
```diff
@@ -153,3 +154,27 @@
     padding-left: 0.5rem;
   }
 }
+
+.p-anchor-link {
+  color: $color-dark;
```
Comment (bartaz) [inline static/sass/styles.scss:159]: Use theme colours
```suggestion
  color: $colors--theme--text-default;
```

File: static/sass/styles.scss [hunk 3]
```diff
@@ -153,3 +154,27 @@
...
+
+  &:visited {
+    color: $color-dark;
```
Comment (bartaz) [inline static/sass/styles.scss:162]:
```suggestion
  color: $colors--theme--text-default;
```

File: static/sass/styles.scss [hunk 4]
```diff
@@ -153,3 +154,27 @@
...
+  }
+
+  &:after {
+    @extend .p-icon--get-link;
+    content: "¶";
```
Comment (bartaz) [inline static/sass/styles.scss:167]: This can probably can be cleaned up now that we are using icon instead.

**Comment by petesfrench** [comment]
Closing as this was just a demo and the changes are now merged in Vanilla

**Pull request closed without merging**